src/ConnectClient/ConnectClient.php

```diff
@@ -19,17 +21,108 @@ class ConnectClient
      */
     public $apiUrl = 'https://connect.givewp.com';
 
+    /**
+     * @unreleased
+     */
+    public function __construct()
+    {
+        if (defined('GIVE_CONNECT_URL') && GIVE_CONNECT_URL) {
+            $this->apiUrl = GIVE_CONNECT_URL;
+        }
+    }
+
     /**
      * Get rest api endpoint url for requests.
      *
-     * @since 2.8.0
+     * @since      2.8.0
+     *
+     * @param string $endpoint The route on the server. E.g: 'gateway-name/connect'
+     */
+    public function getApiUrl(string $endpoint = ''): string
+    {
+        if ( ! empty($endpoint)) {
+            $this->apiUrl .= '/' . ltrim($endpoint, '/');
+        }
+
+        return $this->apiUrl;
+    }
+
+    /**
+     * @unreleased
+     *
+     * @param string $endpoint The route on the server. E.g: 'gateway-name/connect'
+     * @param array  $args     Optional. Can contains 'headers' and 'body'
+     *
+     * @throws Exception
+     */
+    public function get(string $endpoint, array $args = []): array
+    {
+        $url = $this->getApiUrl($endpoint);
+
+        $response = wp_remote_get($url, $args);
+
+        $this->validateWpErrorInResponse($response);
+
+        return $response;
+    }
+
+    /**
+     * @unreleased
+     *
+     * @param string $endpoint The route on the server. E.g: 'gateway-name/connect'
+     * @param array  $args     Optional. Can contains 'headers' and 'body'
+     *
+     * @throws Exception
+     */
+    public function post(string $endpoint, array $args = []): array
+    {
+        $url = $this->getApiUrl($endpoint);
+
+        $response = wp_remote_post($url, $args);
+
+        $this->validateWpErrorInResponse($response);
+
+        return $response;
+    }
+
+    /**
+     * @unreleased
+     *
+     * @param string $endpoint The route on the server. E.g: 'gateway-name/connect'
+     * @param array  $args     Optional. Can contains 'method', 'headers' and 'body'
+     *
+     * @throws Exception
+     */
+    public function request(string $endpoint, array $args = []): array
+    {
+        $url = $this->getApiUrl($endpoint);
+
+        $response = wp_remote_request($url, $args);
+
+        $this->validateWpErrorInResponse($response);
+
+        return $response;
+    }
+
+    /**
+     * @unreleased
      *
-     * @param string $endpoint
+     * @param array|WP_Error $response
      *
-     * @return string
+     * @throws Exception
      */
-    public function getApiUrl($endpoint)
+    private function validateWpErrorInResponse($response)
     {
-        return "{$this->apiUrl}/{$endpoint}";
+        if (is_wp_error($response)) {
+            throw new Exception(
```

Comment:
@glaubersilva Let's make a custom `RequestException` that extends `RuntimeException` so folks can specifically catch this.